src/unix/ftrace.c

```diff
@@ -1211,8 +1239,69 @@ ftrace_send_http_server_error(buffer_handler handler)
         msg_err("ftrace: failed to send HTTP response\n");
 }
 
+
+static boolean
+__ftrace_send_http_chunk_internal(struct ftrace_routine * routine, struct ftrace_printer * p,
+                                  boolean local_printer, buffer_handler out)
+{
+    sysreturn ret;
+    ret = routine->get_fn(p);
+
+    /* no real error handling for http get here */
+    if (ret < 0) {
+        msg_err("%s: get failed with %d\n", __func__, ret);
+        return false;
+    }
+
+    status s = send_http_chunk(out, printer_buffer(p));
+    if (!is_ok(s))
+        goto send_http_chunk_failed;
+
+    if (ret == 0) {
+        s = send_http_chunk(out, 0);
+
+        if (routine->deinit_fn) {
+            /* deinit without init? */
+            assert(routine->init_fn);
+            (void)routine->deinit_fn(p);
+        }
+
+        if (local_printer)
+            printer_deinit(p);
+
+        if (!is_ok(s))
+            goto send_http_chunk_failed;
+        return false;
+    }
+
+    /* reset printer for next chunk */
+    if (printer_init(p, p->flags) < 0) {
+        msg_err("%s: printer_init failed (alloc)\n", __func__);
+        return false;
+    }
+
+    /* ret > 0, so more to send */
+    return true;
+  send_http_chunk_failed:
+    msg_err("%s: send_http_chunk failed with %v\n", __func__, s);
+    return false;
+}
+
+#define SEND_HTTP_CHUNK_INTERVAL_MS     (milliseconds(75))
```

Comment:
Can you explain the reason for this timer here? Why do we need one and why is 75 ms appropriate?

You mention simultaneous requests below, not really sure what that means. Are you saying there are internal
data structures in the http response code that could be corrupted by sending a bunch of chunks one after another?

Reply:
It is kind of kludgey, but I wanted to provide some kind of back-off for the printing and spooling of output buffers. Just running continuously off of the bhqueue or runqueue starves out the draining of queued buffers into lwIP TCP for long periods, leading to a very slow transfer rate until everything is printed. 75ms was selected empirically.

I think we could avoid this kludge once we have the completions in buffer_handlers. But maybe it's ok for now?

As for the comment about simultaneous http requests, just disregard. I didn't realize that it was made exclusive in the inits anyway.